mediapipe/calculators/ovms/openvinoinferencecalculatoroptions.cc

```diff
@@ -0,0 +1,249 @@
+            inputTypes.push_back(tokens[0]);
+        } else {
+            inputTypes.push_back(tag);
+        }
+    }
+
+    for (const auto& [key, value] : tags_to_names) {
+        bool nameMatch = false;
+
+        // Check if supported tag was used
+        for (const auto& supportedTag : supportedTags) {
+            if ( startsWith(key, supportedTag)){
+                if (endsWith(key, "S") && !endsWith(supportedTag, "S"))
+                    continue;
+
+                // Check if used tag is defined in the input_stream
+                for (const auto& graphInput : inputTypes) {
+                    if (startsWith(graphInput, supportedTag)) {
+                        if (endsWith(graphInput, "S") && !endsWith(supportedTag, "S"))
+                            continue;
+                        nameMatch = true;
+                        break;
+                    }
+                }
+            }
+
+            // Check if empty tag used - OV:Tensor default type
+            if ( tokenize(key, ':').size() == 0) {
+                // Check if used tag is defined in the input_stream
+                for (const auto& graphInput : inputTypes) {
+                    // Check if empty tag used - OV:Tensor default type
+                    if (key == graphInput) {
+                        nameMatch = true;
+                        break;
+                    }                       
+                }
+            }
+
+            // Type used in tag_to__tensor_names does match input_stream type
+            if (nameMatch){
+                break;
+            }
+        }
+
+        // Check if no supported tag used - OV:Tensor default type
+        for (const auto& graphInput : inputTypes) {
+            // Full match required
+            if (key == graphInput) {
+                nameMatch = true;
+                break;
+            }
+        }
+
+        // Type used in tag_to__tensor_names does not match outputstream type
+        if (!nameMatch)
+        {
+            LOG(ERROR) << "OpenVINOInferenceCalculator GetContract error. Stream names mismatch for tag_to__tensor_names name key " << key;
+            return false;
+        } 
+    }
+
+    return true;
+}
+
+bool ValidateOptions(CalculatorContract* cc) {
+    const auto& options = cc->Options<OpenVINOInferenceCalculatorOptions>();
+
+    if (options.tag_to_output_tensor_names().size() > 0 && options.output_order_list().size() > 0) {
+        LOG(ERROR) << "OpenVINOInferenceCalculator GetContract error. Use tag_to_output_tensor_names or output_order_list not both at once.";
+        return false;
+    }
+
+    if (options.tag_to_input_tensor_names().size() > 0 && options.input_order_list().size() > 0) {
+        LOG(ERROR) << "OpenVINOInferenceCalculator GetContract error. Use tag_to_input_tensor_names or input_order_list not both at once.";
+        return false;
+    }
+
+    return true;
+}
+
+bool ValidateCalculatorSettings(CalculatorContract* cc)
```

Comment:
Only the one exposed in header and used in calculator should not be static, all other should be flagged as static